packages/backend/src/entity/User.ts

```diff
@@ -34,6 +34,10 @@ export class User {
     events: Array<Event>;
 
     async hashPassword(password: string) {
-        this.passwordHash = await bcrypt.hash(password, costFactor);
```

Comment:
I think the problem comes from the src/config/config.ts file where we must change the line 37 to parseInt the env variables.
Otherwise, I don't think it's necessary to modify the User.ts entity file

Reply:
I've [read](https://stackoverflow.com/a/43092310/6630397) it was about this 'salt' variable that has to be passed this way for `bcrypt` to work properly, otherwise the original error message shows up. Maybe it's also possible to make it work with two `await` nested statements but I didn't succeed to make it work this way.
Then I also figured out it was interpreting the cost factor environment variable as a whole string instead of an int (vscode actually told me that it was not formatted properly).
The actual commit is actually working for me and should not impact the entity.

I'm also pushing a news commit to remove the `console.log()` statements.

Reply:
As you can see in the documentation https://www.npmjs.com/package/bcrypt#to-hash-a-password, there is two ways to generate a hash. Currently we're using the second one, but if you switch to the second one, it's possible that the current login on the api will not work anymore because of the change of salt. So it would be better if you don't change that. Maybe if as I say you use Number.parseInt on the line 37 of the src/config/config.ts it will resolve your problem.

Reply:
But I'm guessing the initial error is due to the fact that the second option (the one we have chosen) is somehow not valid (I don't know why actually, but if someone can figure it out and fix it, that's ok as well).

Reply:
It is invalid when the value comes from the environment variable because the value of an environment variable is always a string when retrieved. It must be parsed before being used when another type is needed. Since it is a string, bcrypt believes it to be a salt and not a cost factor (I assume its implementation automagically uses the second argument as a salt if it's a string, or as the cost factor if it's a number).

The way you fixed it here is not the best idea. You are parsing it at the point of use instead of parsing it when it is loaded in the configuration. If we ever needed to use the same configuration variable elsewhere, we would need to parse it everywhere it is used if we do it like this. What we should do is parse environment variables in `config.ts` before they are used.

I have copied some parsing functions from another project and suggested a change in https://github.com/MediaComem/modos/pull/69. I also took the liberty of validating booleans, port numbers and URLs since I had the functions at hand.

If you agree with these changes, you may merge my PR into this one and then merge this one into master.